src/Square3x3.java

```diff
@@ -0,0 +1,206 @@
+        }
+    }
+
+    /**
+     * Returns a string representation of the Square3x3 object
+     *
+     * @return a string represent the Square3x3 object, For example:
+     * 1   2   3
+     * 4   5   6
+     * 7   8   9
+     */
+    public String toString() {
+        String msg = "";
+        for (int i = 0; i < SIZE; i++) {
+            for (int j = 0; j < SIZE; j++) {
+                msg += square[i][j];
+                if (j < SIZE - 1)
+                    msg += "\t";
+            }
+            msg += "\n";
+        }
+        return msg;
+    }
+
+    // print a row in square
+    //TODO: delete - only for tests
+    public void printRow(int row) {
+        for (int col = 0; col < SIZE; col++)
+            System.out.print(getCell(row, col) + "\t");
+    }
+
+    /**
+     * Returns true if all numbers 1-9 exists in square
+     *
+     * @return true if all numbers 1-9 exists in square
+     */
+    public boolean allThere() {
+        initWhichNumbersAppear(); // init WHICH_NUMBERS_APPEAR boolean array
+
+        // go over all square rows
+        for (int i = 0; i < SIZE; i++) {
+            this.whosThereRow(i, WHICH_NUMBERS_APPEAR); // check whosThereRow for each row
+        }
+
+        // go over WHICH_NUMBERS_APPEAR boolean array
+        for (int j = 0; j < WHICH_NUMBERS_APPEAR.length; j++) {
+            if (!WHICH_NUMBERS_APPEAR[j])
+                return false; // a number between 1-9 is not in square
+        }
+        return true; // numbers 1-9 are all in square
+    }
+
+    /**
+     * Check if square row contains cells with values between 1-9
+     * If yes - update values boolean array item in the corresponding index to true
+     * For example:
+     * Row 1 cells are: 10  7   1
+     * whosThereRow will update values[7] and values[1] to true
+     *
+     * @param row    - row in square to check
+     * @param values - boolean array to update correspondingly to cells in row
+     */
+    public void whosThereRow(int row, boolean[] values) {
+        // check if values is not null and row param is valid index
+        if (values != null && validIndex(row)) {
+            // go over row's columns
+            for (int i = 0; i < SIZE; i++) {
+                if (square[row][i] > 0 && square[row][i] < values.length) // check if cell value is between 1-9
+                    values[square[row][i]] = true; // update values array item in the corresponding index to true
+            }
+        }
+    }
+
+    /**
+     * Check if square column contains cells with values between 1-9
+     * If yes - update values boolean array item in the corresponding index to true
+     * For example:
+     * Column 1 cells are: 10  7   1
+     * whosThereCol will update values[7] and values[1] to true
+     *
+     * @param col    - column in square to check
+     * @param values - boolean array to update correspondingly to cells in row
+     */
+    public void whosThereCol(int col, boolean[] values) {
```

Comment:
actually, you could create a new method: "whosThere" that accepts a number, a flag (boolean) and  the `values` array.
Then you could calculate whosThere based on the flag, so it ends up like that:
 ```
public void whoThereCol(int col. boolean[] values) {
   whosThere(col, false, values)
}

public void whoThereRow(int row. boolean[] values) {
   whosThere(col, true, values)
}

public void whosThere(int index, boolean isRow, boolean[] values) {
...
   switch (isRow):
...
}